src/controllers/users.ts

```diff
@@ -1,18 +1,78 @@
 import User from '../models/user';
-import { CreateUserParams } from '../interfaces';
+import { CreateUserParams, UpdateUserParams } from '../interfaces';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
 
 export const getUsers = async () => await User.findAll();
 
-export const getUserById = async (userId: number) => await User.findOne({ where: { id: userId } });
+export const getUserById = async (userId: number) =>
+  await User.findOne({
+    where: { id: userId },
+  });
+
+const createToken = (id: number, email: string) =>
+  jwt.sign({ id, email }, process.env.ACCESS_TOKEN_SECRET || 'TOP_SECRET');
+
+export const createUser = async (createUserParams: CreateUserParams) => {
+  console.log('hola');
+  const hashedPassword = await bcrypt.hash(createUserParams.password, 10);
+  createUserParams.password = hashedPassword;
+
+  const user = await User.findOne({
+    where: { email: createUserParams.email },
+  });
+
+  if (user) {
+    throw new Error('The user already exists');
+  }
+
+  return await User.create(createUserParams);
+};
```

Comment:
I think we can rename this to register instead, also, let's remove the console.log